### PR TITLE
login-screen

### PR DESCRIPTION
**Pull request by yashkapoor8710**
Screenshot 2020-03-04 at 7 35 15 PM

### CODE REVIEW

**Comment by yashkapoor8710** [comment]
Screenshot 2020-03-05 at 1 55 58 PM

File: router/index.tsx [hunk 1]
```diff
@@ -25,7 +26,8 @@ const Router: React.FunctionComponent<Props> = ({
         <NativeRouter>
           <BackHandlerHOC>
             <Switch>
-              <Route exact path="/" component={ChatList} />
+              <Route exact path="/" component={Login} />
+              <Route exact path="/login" component={ChatList} />
```
Comment (manojsinghnegiwd) [inline router/index.tsx]: Change path to `/chatList`

File: components/Chat/LoginScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    const theme: AppTheme = useTheme();
+
+    return (
+        <View>
+            <View style={style.container}>
+                <ChatUserImage
+                    source={userImageSource}
+                    containerStyle={style.userImageContainer}
+                    imageStyle={{width: 100, height: 100, borderRadius: 50}}
+                />
```
Comment (manojsinghnegiwd) [inline components/Chat/LoginScreen.tsx]: Use normal Image here

File: router/index.tsx (hunk 1 above)
Comment (yashkapoor8710) [reply router/index.tsx]: I have chnaged the path

File: components/Chat/LoginScreen.tsx (hunk 2 above)
Comment (yashkapoor8710) [reply components/Chat/LoginScreen.tsx]: I have used the Image

File: components/Chat/PlaceholderField.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    const theme: AppTheme = useTheme();
+
+    return (
+        <View>
+            <View style={[style.containerNew, {borderColor: theme.lightBottomColor}]}>
+                <TextInput
+                    placeholder={firstPlaceholder}
+                    placeholderTextColor={theme.lightTextColor}
+                    style={{ color: theme.textColor }}
+                />
```
Comment (manojsinghnegiwd) [inline components/Chat/PlaceholderField.tsx]: I think this should be created as a seprate component

File: pages/Login/index.tsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+                    source={{ uri: constants.appLogo }}
+                    style={{width: 100, height: 100, borderRadius: 50}}
+                />
+            </View>
+            <View style={[style.contentContainer ,{paddingTop: 10, paddingBottom: 30}]}>
+                <ThemedText styleKey="textColor" style={style.nameStyle}>{constants.appName}</ThemedText>
+            </View>
+            <PlaceholderField
+                firstPlaceholder={constants.usernamePlacerHolder}
+                secondPlaceholder={constants.passwordPlacerHolder}
```
Comment (manojsinghnegiwd) [inline pages/Login/index.tsx]: Again the same problem will be able to resue this in signup? even if we will be able to what about if the signup page doesn't take username?

File: pages/Login/index.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+            />
+             <View style={[style.container,{paddingTop: 50}]}>
+                <TouchableOpacity onPress={goToChatList} style={[style.loginStyle, {borderColor: theme.lightBottomColor}]}>
+                    <ThemedText styleKey="textColor">Login</ThemedText>
+                    <MaterialIcon name="login" size={20} color={theme.textColor} style={{paddingTop: 3}}/>
```
Comment (manojsinghnegiwd) [inline pages/Login/index.tsx]: Remove this Icon

Comment (yashkapoor8710) [reply pages/Login/index.tsx]: Removed the icon

File: pages/Login/index.tsx (hunk 4 above)
Comment (yashkapoor8710) [reply pages/Login/index.tsx]: Modified the Code

File: components/Chat/PlaceholderField.tsx (hunk 3 above)
Comment (yashkapoor8710) [reply components/Chat/PlaceholderField.tsx]: I have made a separate component

**Comment by yashkapoor8710** [comment]
I have fixed the PR Issue just validation left

File: components/Chat/ErrorText.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { StyleSheet, View, TextStyle } from 'react-native';
+import { AppTheme } from '../../config/DefaultConfig';
+import useTheme from "../../hooks/useTheme";
+import ThemedText from '../UI/ThemedText';
+
+interface Props {
+    error: any;
```
Comment (manojsinghnegiwd) [inline components/Chat/ErrorText.tsx]: Define proper type

File: components/Chat/ErrorText.tsx [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+const ErrorText: React.FunctionComponent<Props> = ({
+    error,
+}: Props) => {
+    const theme: AppTheme = useTheme();
+
+    return (
+        <View>
+            {
+                error && error.length ? 
+                    <ThemedText styleKey="warningColor" style={style.textContainer}>{error[0]}</ThemedText>
```
Comment (manojsinghnegiwd) [inline components/Chat/ErrorText.tsx]: name the styleKey to `errorColor`

File: components/Chat/Input.tsx [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+import { isIOS } from '../../utils';
+import ErrorText from './ErrorText';
+
+interface Props {
+    placeholder: string;
+    value?: string;
+    onChange?: any;
+    secureCheck?: any;
+    validation?: any;
+};
```
Comment (manojsinghnegiwd) [inline components/Chat/Input.tsx]: This is not how it should be we should have extended the `TextInputProps` to add new props

File: components/Chat/Input.tsx [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+import useTheme from "../../hooks/useTheme";
+import { isIOS } from '../../utils';
+import ErrorText from './ErrorText';
+
+interface Props {
+    placeholder: string;
+    value?: string;
+    onChange?: any;
+    secureCheck?: any;
+    validation?: any;
```
Comment (manojsinghnegiwd) [inline components/Chat/Input.tsx]: Prop name should be `errors`

File: components/Chat/Input.tsx [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+        padding: 10,
+        paddingTop: isIOS() ? 10 : 0,
+        paddingBottom: isIOS() ? 0 : 0,
+        marginTop: 10,
+        borderBottomWidth: 2,
+    },
+    textContainer: {
+        flex: 9,
+        paddingTop: isIOS() ? 10 : 0,
+        paddingBottom: isIOS() ? 10 : 0,
```
Comment (manojsinghnegiwd) [inline components/Chat/Input.tsx]: If this is reused 4 times please put it in a variable

File: pages/Login/index.tsx [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+                <TouchableOpacity onPress={goToChatList} style={[style.loginStyle, {borderColor: theme.lightBottomColor}]}>
+                    <ThemedText styleKey="textColor">{constants.loginButton}</ThemedText> 
+                </TouchableOpacity> 
+            </View>
+            </KeyboardAvoidingView>
+        </View>
+        <View style={style.topContainer}>
+            <ThemedText styleKey="lightTextColor">{constants.signupCheck}</ThemedText>
+            <TouchableOpacity>
+                <ThemedText styleKey="lightTextColor" style={{color:'#ed5d45'}}>{constants.signupButton}</ThemedText>
```
Comment (manojsinghnegiwd) [inline pages/Login/index.tsx]: Please remove hard coded color

File: pages/Login/index.tsx [hunk 12]
```diff
@@ -0,0 +1,171 @@
...
+        else {
+            setErrors(errors)
+        }
+    }
+
+    const constants: AppConstants = useConstants();
+    const theme: AppTheme = useTheme();
+    const keyboardVerticalOffset = isIOS ? 40 : 0;
+
+    console.log(errors, 'validError')
```
Comment (manojsinghnegiwd) [inline pages/Login/index.tsx]: remove console

File: components/Chat/Input.tsx (hunk 8 above)
Comment (yashkapoor8710) [reply components/Chat/Input.tsx]: I have extended the TypeInputProps

File: components/Chat/ErrorText.tsx (hunk 7 above)
Comment (yashkapoor8710) [reply components/Chat/ErrorText.tsx]: I have changed the color name

File: components/Chat/Input.tsx (hunk 9 above)
Comment (yashkapoor8710) [reply components/Chat/Input.tsx]: I have changed the prop name

File: components/Chat/ErrorText.tsx (hunk 6 above)
Comment (yashkapoor8710) [reply components/Chat/ErrorText.tsx]: I have defined the prop type

File: pages/Login/index.tsx (hunk 11 above)
Comment (yashkapoor8710) [reply pages/Login/index.tsx]: I have moved the app color

File: pages/Login/index.tsx (hunk 12 above)
Comment (yashkapoor8710) [reply pages/Login/index.tsx]: I have removed the console

File: components/Chat/Input.tsx (hunk 10 above)
Comment (yashkapoor8710) [reply components/Chat/Input.tsx]: I have used a variable for common padding

**Pull request merged**